src/samplers/powerspherical.jl

```diff
@@ -0,0 +1,37 @@
+# Sampler for Power Spherical
+
+struct PowerSphericalSampler{T <: Real}
+    μ::Vector{T}
+    κ::T
+    d::Int
+    dist_b::Beta
+    dist_u::HyperSphericalUniform
+end
+
+function _rand!(rng::AbstractRNG, spl::PowerSphericalSampler, x::AbstractVector)
+    z = rand(rng, spl.dist_b)
+    v = rand(rng, spl.dist_u)
+    t = 2 * z - 1
+    m = sqrt(1 - t ^ 2) * v'
+    y = [t; m]
+    e_1 = [1.; zeros(eltype(spl.μ), spl.d -1)]
+    u = e_1 - spl.μ
+    normalize!(u)
+    x .= (-1) * (Matrix{eltype(spl.μ)}(I, spl.d, spl.d) .- 2*u*u') * y
+end
+
+function _rand(rng::AbstractRNG, spl::PowerSphericalSampler, x::AbstractVector)
```

Comment:
Is this method needed? Or does the default call the `_rand!` method above after allocating `x`?

Reply:
I need Zygote to diff through `rand(PowerSpherical)` (so a non-array mutating version would be nice) but I just realised that that only works if you call `rand` without providing an RNG. Maybe we could have a `rsample()` method or something alike?

Reply:
I have to correct myself, when correctly implementing a non-mutating `rand()` function, Zygote knows how to handle this. I agree that a `ChainRule` (and potentially a `DiffRule`) would probably be best, but I just don't have time for that at the moment. Can we just have a non-mutating `rand()` version for know?